recipes/qt/all/conanfile.py

```diff
@@ -0,0 +1,676 @@
+            if config.has_option(section, "depends"):
+                res[modulename]["depends"] = [str(i) for i in config.get(section, "depends").split()]
+    return res
+
+
+class QtConan(ConanFile):
+    _submodules = _getsubmodules()
+
+    generators = "pkg_config"
+    name = "qt"
+    description = "Qt is a cross-platform framework for graphical user interfaces."
+    topics = ("conan", "qt", "ui")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.qt.io"
+    license = "LGPL-3.0"
+    exports = ["qtmodules.conf", "patches/*.diff"]
+    settings = "os", "arch", "compiler", "build_type"
+
+    options = {
+        "shared": [True, False],
+        "commercial": [True, False],
+
+        "opengl": ["no", "es2", "desktop", "dynamic", "auto"],
+        "with_vulkan": [True, False],
+        "openssl": [True, False],
+        "with_pcre2": [True, False],
+        "with_glib": [True, False],
+        # "with_libiconv": [True, False],  # QTBUG-84708 Qt tests failure "invalid conversion from const char** to char**"
+        "with_doubleconversion": [True, False],
+        "with_freetype": [True, False],
+        "with_fontconfig": [True, False],
+        "with_icu": [True, False],
+        "with_harfbuzz": [True, False],
+        "with_libjpeg": [True, False],
+        "with_libpng": [True, False],
+        "with_sqlite3": [True, False],
+        "with_mysql": [True, False],
+        "with_pq": [True, False],
+        "with_odbc": [True, False],
+        "with_libalsa": [True, False],
+        "with_openal": [True, False],
+        "with_zstd": [True, False],
+
+        "GUI": [True, False],
+        "widgets": [True, False],
+
+        "device": "ANY",
+        "cross_compile": "ANY",
+        "sysroot": "ANY",
+        "config": "ANY",
+        "multiconfiguration": [True, False],
+    }
+    options.update({module: [True, False] for module in _submodules if module != 'qtbase'})
+
+    no_copy_source = True
+    default_options = {
+        "shared": False,
+        "commercial": False,
+        "opengl": "auto",
+        "with_vulkan": False,
+        "openssl": True,
+        "with_pcre2": True,
+        "with_glib": False,
+        # "with_libiconv": True, # QTBUG-84708
+        "with_doubleconversion": True,
+        "with_freetype": True,
+        "with_fontconfig": True,
+        "with_icu": True,
+        "with_harfbuzz": False,
+        "with_libjpeg": True,
+        "with_libpng": True,
+        "with_sqlite3": True,
+        "with_mysql": True,
+        "with_pq": True,
+        "with_odbc": True,
+        "with_libalsa": False,
+        "with_openal": True,
+        "with_zstd": True,
+
+        "GUI": True,
+        "widgets": True,
+
+        "device": None,
+        "cross_compile": None,
+        "sysroot": None,
+        "config": None,
+        "multiconfiguration": False,
+    }
+    default_options.update({module: False for module in _submodules if module != 'qtbase'})
```

Comment:
does it make sense to enable qt essential modules by default ?

Reply:
it would, but unfortunately it increases the build time so much, that it will probably be too much for C3I

Reply:
any idea what is the build time limitation on C3I? I will try to build it in our CI to get some statistics

Reply:
@uilianries how much time is allowed for a recipe to build ?

Reply:
it tooks me 43 minutes to build one configuration with all essential modules enabled on an aws c5.2xlarge instance.